File: onyx-database-tests/test/remote/exception/TestCallbackExceptions.java
package remote.exception;

import com.onyx.exception.EntityCallbackException;
import com.onyx.exception.EntityException;
import com.onyx.exception.InitializationException;
import org.junit.After;
import org.junit.Before;
import org.junit.Test;
import org.junit.experimental.categories.Category;
import remote.base.RemoteBaseTest;
import entities.exception.EntityCallbackExceptionEntity;

import java.io.IOException;
import category.RemoteServerTests;


/**
 * Created by timothy.osborn on 2/10/15.
 */
@Category({ RemoteServerTests.class })
public class TestCallbackExceptions extends RemoteBaseTest
{

    /**
     * Created by timothy.osborn on 12/14/14.
     */
    @Before
    public void before() throws InitializationException
    {
        initialize();
    }

    @After
    public void after() throws EntityException, IOException
    {
        shutdown();
    }

    int z = 0;

    protected synchronized void increment()
    {
        z++;
    }

    protected synchronized int getZ()
    {
        return z;
    }

    @Test(expected = EntityCallbackException.class)
    public void testPersistCallbackException() throws EntityException
    {
        EntityCallbackExceptionEntity entity = new EntityCallbackExceptionEntity();
        manager.saveEntity(entity);
    }

}

File: onyx-database-tests/test/remote/exception/TestCallbackExceptions.kt
package remote.exception

import category.RemoteServerTests
import com.onyx.exception.EntityCallbackException
import com.onyx.exception.OnyxException
import com.onyx.exception.InitializationException
import com.onyx.persistence.IManagedEntity
import entities.exception.EntityCallbackExceptionEntity
import org.junit.After
import org.junit.Before
import org.junit.Test
import org.junit.experimental.categories.Category
import remote.base.RemoteBaseTest

import java.io.IOException


/**
 * Created by timothy.osborn on 2/10/15.
 */
@Category(RemoteServerTests::class)
class TestCallbackExceptions : RemoteBaseTest() {

    /**
     * Created by timothy.osborn on 12/14/14.
     */
    @Before
    @Throws(InitializationException::class)
    fun before() {
        initialize()
    }

    @After
    @Throws(IOException::class)
    fun after() {
        shutdown()
    }

    @get:Synchronized private var z = 0

    @Synchronized private fun increment() {
        z++
    }

    @Test(expected = EntityCallbackException::class)
    @Throws(OnyxException::class)
    fun testPersistCallbackException() {
        val entity = EntityCallbackExceptionEntity()
        manager!!.saveEntity<IManagedEntity>(entity)
    }

}
